iset.mm

```diff
@@ -69544,6 +69545,49 @@ elements or fails to hold (is equal to ` (/) ` ) for some element.
     VFVHVGUTVFPZVHVGFVAOIJZVIVAVDOSVJVFUTVJVBOHIJZUHVJOVAIJZVBVKPVAOUIVLVBVKOVA
     HSTRUAUBRTUOUJUKULUMURUSVCVBABCDUPUNUQ $.
 
+  ${
+    $d A r s $.  $d A z $.  $d ph x $.  $d s x $.
+    exmidonfinlem.a $e |- A
+      = { { x e. { (/) } | ph } , { x e. { (/) } | -. ph } } $.
+    $( Lemma for ~ exmidonfin .  (Contributed by Andrew W Swan and Jim Kingdon,
+       9-Mar-2024.) $)
+    exmidonfinlem $p |- ( _om = ( On i^i Fin ) -> DECID ph ) $=
+      ( vz vs vr vw vy com wceq wcel c0 cv wral eleq2 sylib wb cvv c2o con0 cfn
+      cin csn crab wn wo wdc word wtr cpr elpri eleq2s elrabi velsn biidd elrab
+      biimpcd simprbi notnotd wex 0ex r19.3rmv ax-mp rabeq0 sylibr eqtr4d rabex
+      p0ex prid2 eleqtrri syl6eqel prid1 mpan9 rgen2 dftr5 mpbir ordtriexmidlem
+      snm syl6 jaod ontrci treq mpbiri jaoi syl rgen dford3 prexg mp2an eqeltri
+      mpbir2an elon cen wbr 2onn nnfi wne pm5.19 mtbi rabbi pr2ne eqbrtri enfii
+      neir elini c1o df1o2 1lt2o eqeltrri nneneq mpan2 eleqtrrid wa snid simprd
+      mpbii orim12i df-dc 3syl ) JUAUBUCZKZCJLZMUDZABYDUEZKZYDAUFZBYDUEZKZUGZAU
+      HZYBYCCYALCUAUBCUALCUIZYLCUJZENZUJZECOYMFNZCLZFGNZOGCOYQGFCYRYRCLYRYEKZYR
+      YHKZUGZYPYRLZYQUUAYRYEYHUKZCYRYEYHULDUMUUBYSYQYTUUBYSYPYELZYQYSUUBUUDYRYE
+      YPPURUUDYPYHCUUDYPMYHUUDYPYDLZYPMKZABYPYDUNFMUOZQUUDYGUFZBYDOZYHMKUUDUUHU
+      UIUUDAUUDUUEAAABYPYDBNZYPKZAUPUQUSUTHNYDLHVAZUUHUUIRHMVBVSZUUHBHYDVCVDQYG
+      BYDVEVFVGYHUUCCYEYHYGBYDVIVHZVJDVKVLVTUUBYTYPYHLZYQYTUUBUUOYRYHYPPURUUOYP
+      YECUUOYPMYEUUOUUEUUFYGBYPYDUNUUGQUUOYGBYDOZYEMKUUOYGUUPUUOUUEYGYGYGBYPYDU
+      UKYGUPUQUSUULYGUUPRUUMYGBHYDVCVDQABYDVEVFVGYEUUCCYEYHABYDVIVHZVMDVKVLVTWA
+      VNVOGFCVPVQYOECYNCLYNYEKZYNYHKZUGZYOUUTYNUUCCYNYEYHULDUMUURYOUUSUURYOYEUJ
+      YEABVRWBYNYEWCWDUUSYOYHUJYHYGBVRWBYNYHWCWDWEWFWGECWHWLCCUUCSDYESLZYHSLZUU
+      CSLUUQUUNYEYHSSWIWJWKWMVQTUBLZCTWNWOZCUBLTJLZUVCWPTWQVDCUUCTWNDUUCTWNWOZY
+      EYHWRZYEYHAYGRZBYDOZYEYHKUVHUVIAWSINYDLIVAUVHUVIRIMVBVSUVHBIYDVCVDWTAYGBY
+      DXAWTXEUVAUVBUVFUVGRUUQUUNYEYHSSXBWJVQXCZCTXDWJXFJYACPWDYCYDCLYJYCYDTCXGY
+      DTXHXIXJYCUVDCTKZUVJYCUVEUVDUVKRWPCTXKXLXQXMYJYDUUCCYDYEYHULDUMWFYJAYGUGY
+      KYFAYIYGYFMYDLZAYFMYELZUVLAXNYFUVLUVMMVBXOZYDYEMPXQAABMYDUUJMKZAUPUQQXPYI
+      UVLYGYIMYHLZUVLYGXNYIUVLUVPUVNYDYHMPXQYGYGBMYDUVOYGUPUQQXPXRAXSVFXT $.
+  $}
+
+  ${
+    $d x z $.
+    $( If a finite ordinal is a natural number, excluded middle follows.  The
+       converse should also be provable as shown in the Metamath Proof
+       Explorer.  (Contributed by Andrew W Swan and Jim Kingdon,
+       9-Mar-2024.) $)
+    exmidonfin $p |- ( _om = ( On i^i Fin ) -> EXMID ) $=
```

Comment:
Looking at the informal proof, it looks like `|- ( ( On i^i Fin ) C_ _om -> EXMID )` ?  If that's correct, then I think that statement is clearer. And I presume that  `|- _om C_ ( On i^i Fin )` ? It's probably already in iset, I haven't looked; there should be a link to it from this theorem.

Reply:
Yeah, it is worth clearing up the comment on this point. I've removed the word "converse" in favor of explicitly talking about excluded middle. And I have linked to the theorems you describe about a natural number being a finite ordinal ( https://us.metamath.org/ileuni/nnfi.html and https://us.metamath.org/ileuni/nnon.html ).

As for the statement of the theorem, it is chosen to match https://us.metamath.org/mpeuni/onfin2.html